component/discovery/discovery.go

```diff
@@ -12,19 +14,91 @@ import (
 // component.
 type Target map[string]string
 
-// maxUpdateFrequency is the minimum time to wait between updating targets.
-// Currently not settable, since prometheus uses a static threshold, but
-// we could reconsider later.
-const maxUpdateFrequency = 5 * time.Second
+// Exports holds values which are exported by all discovery components.
+type Exports struct {
+	Targets []Target `river:"targets,attr"`
+}
 
 // Discoverer is an alias for Prometheus' Discoverer interface, so users of this package don't need
 // to import github.com/prometheus/prometheus/discover as well.
 type Discoverer discovery.Discoverer
 
-// RunDiscovery is a utility for consuming and forwarding target groups from a discoverer.
+// DiscoveryCreator is a function provided by an implementation to create a static Discoverer instance
+type DiscoveryCreator func(component.Arguments, component.Options) (Discoverer, error)
+
+type DiscoveryComponent struct {
```

Comment:
This stutters a little since the package name is already discovery. Rename to `Component`?